analyzere/base_resources.py

```diff
@@ -148,6 +149,13 @@ def to_dict(self):
             d[k] = to_dict(v)
         return d
 
+    def __hash__(self):
+        # Certainly not the most efficient hash, since anything without an ID
+        # has the same hash value.
+        if hasattr(self, "id"):
+            int(uuid.UUID(self.id))
+        return 0
```

Comment:
Worst case scenario, objects without an ID hash to the same value, and are therefore treated like a list (bad lookup time, locally).

I'm certainly open to implementing a more robust implementation. What fields do you think are relevant to hash? Particularly interested in how to handle nested structures, since the abstraction makes these almost seamless.

It's worth considering the effect it will have on lazy loading.